### DISCUSSION TITLE
I'm using the OpenTelemetry Java agent and want to remove resource information when collecting logs.

### DISCUSSION THREAD

**Discussion by JangCool** [open-telemetry/opentelemetry-java-instrumentation#11859] [Q&A] [answered]
I'm receiving log information using the Java collector server version, but I do not want to receive resource information.

Even if I pass resource-related settings as empty values, they are still sent. Are they being sent as default values? Is there a way to exclude them? It seems to unnecessarily consume data for transmission. The same goes for trace and metrics information. Please help.

otel.resource.attributes=
otel.java.disabled.resource.providers=

```
{
  "resource": {
    "attributes": [
      {
        "key": "service.instance.id",
        "value": {
          "string_value": "8d897dbf-9002-484a-ad37-bfac06b898f7"
        }
      },
      {
        "key": "service.name",
        "value": {
          "string_value": "hamonica2-api"
        }
      },
      {
        "key": "telemetry.distro.name",
        "value": {
          "string_value": "opentelemetry-java-instrumentation"
        }
      },
      {
        "key": "telemetry.distro.version",
        "value": {
          "string_value": "2.6.0"
        }
      },
      {
        "key": "telemetry.sdk.language",
        "value": {
          "string_value": "java"
        }
      },
      {
        "key": "telemetry.sdk.name",
        "value": {
          "string_value": "opentelemetry"
        }
      },
      {
        "key": "telemetry.sdk.version",
        "value": {
          "string_value": "1.40.0"
        }
      }
    ]
  },
  "scope_logs": [
    {
      "scope": {
        "name": "org.apache.catalina.core.ContainerBase.[Tomcat].[localhost].[/]"
      },
      "log_records": [
        {
          "time_unix_nano": 1721348753770000000,
          "severity_number": "SEVERITY_NUMBER_INFO",
          "severity_text": "INFO",
          "body": {
            "string_value": "Initializing Spring DispatcherServlet 'dispatcherServlet'"
          },
          "flags": 1,
          "trace_id": "\366\232\271\236`\322\211\006\347U\325\344\214\'\347f",
          "span_id": "xG\214\317EK\233\304",
          "observed_time_unix_nano": 1721348753770651800
        }
      ]
    },
    {
      "scope": {
        "name": "kr.co.pionnet.hamonica2.zconfiguration.interceptor.HamonicaLocaleInterceptor"
      },
      "log_records": [
        {
          "time_unix_nano": 1721348753797000000,
          "severity_number": "SEVERITY_NUMBER_DEBUG",
          "severity_text": "DEBUG",
          "body": {
            "string_value": "locale : ko"
          },
          "flags": 1,
          "trace_id": "\366\232\271\236`\322\211\006\347U\325\344\214\'\347f",
          "span_id": "xG\214\317EK\233\304",
          "observed_time_unix_nano": 1721348753797680700
        }
      ]
    }
  ]
}


```

**Comment by laurit** [comment]
Resource information is sent once per exported batch so when your exported batch has 100 spans there will be only a singe copy of the resource attributes.

> otel.java.disabled.resource.providers=

you will need to list the class names of the providers you wish to disable. Most of the providers can be found in https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/resources/library

**Reply by JangCool** [reply]
Even though I have added all the resources to the list as shown below, the issue still persists. How can I remove properties that start with the [telemetry.]? And is service.instance a default value?



otel.java.disabled.resource.providers=io.opentelemetry.instrumentation.resources.ContainerResourceProvider,io.opentelemetry.instrumentation.resources.HostResourceProvider,io.opentelemetry.instrumentation.resources.OsResourceProvider,io.opentelemetry.instrumentation.resources.ProcessResourceProvider,io.opentelemetry.instrumentation.resources.ProcessRuntimeResourceProvider,io.opentelemetry.instrumentation.resources.ManifestResourceProvider,io.opentelemetry.instrumentation.resources.HostIdResourceProvider,io.opentelemetry.instrumentation.resources.HostIdResourceProvider,io.opentelemetry.instrumentation.resources.AttributeResourceProvider

**Reply by laurit** [reply, marked as answer]
The attributes starting with `telemetry` are required by otel specification https://opentelemetry.io/docs/specs/semconv/resource/#telemetry-sdk As far as I know [opentelemetry java sdk](https://github.com/open-telemetry/opentelemetry-java) does not offer a configuration option for disabling these. If you need to remove this I'd try implementing an [agent extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) Have a look at https://github.com/signalfx/splunk-otel-java/blob/main/custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java for ideas, if you wish to drop all resource attributes you could return `Resource.empty()` from the customizer.
`service.instance.id` is provided by https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/resources/ServiceInstanceIdResourceProvider.java

**Reply by JangCool** [reply]
I understand. Thank you for the answer.

**Reply by ogerardin** [reply]
I have a similar need. What did you end up doing?

I tried to set:

`OTEL_JAVA_ENABLED_RESOURCE_PROVIDERS=io.opentelemetry.sdk.autoconfigure.EnvironmentResourceProvider`

But the result is that it completely stops traces from being transmitted...

So now I just remove the keys I'm less interested in with `OTEL_RESOURCE_DISABLED_KEYS `but it would be better if I could disable resource collection altogether (except the mandatory part).
